File: Week-7/School_Management_System/backend/src/main/java/com/studentmanagement/service/AttendanceService.java
package com.studentmanagement.service;

import com.studentmanagement.dto.attendance.AttendanceRequest;
import com.studentmanagement.dto.attendance.AttendanceResponse;

import java.util.List;

public interface AttendanceService {
    AttendanceResponse mark(Long studentId, AttendanceRequest request);
    List<AttendanceResponse> getForStudent(Long studentId);
    List<AttendanceResponse> getForCurrentStudent();
}

File: Week-7/School_Management_System/frontend/src/app/services/attendance.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Attendance, AttendanceRequest } from '../models/attendance.model';

@Injectable({ providedIn: 'root' })
export class AttendanceService {
  private readonly http = inject(HttpClient);

  getForStudent(studentId: number): Observable<Attendance[]> {
    return this.http.get<Attendance[]>(`${environment.apiUrl}/attendance/students/${studentId}`);
  }

  getMine(): Observable<Attendance[]> {
    return this.http.get<Attendance[]>(`${environment.apiUrl}/attendance/me`);
  }

  mark(studentId: number, request: AttendanceRequest): Observable<Attendance> {
    return this.http.put<Attendance>(`${environment.apiUrl}/attendance/students/${studentId}`, request);
  }
}
